pkgs/build-support/autoconf-cache/default.nix

```diff
@@ -0,0 +1,18 @@
+{ stdenv, gettext, autoconf, automake }:
+
+stdenv.mkDerivation {
+  name = "autoconf-cache";
+  src = ./src;
+
+  buildInputs = [ gettext autoconf automake ];
+  buildPhase = ''
+    autoreconf --install --force
+    echo 'declare -p > seed.1' >> configure
+    ./configure
+    grep -E -- '-- (ac|am|gl)_cv_' < seed.1 | sed 's/declare --/export/' > seed.2
```

Comment:
i see the bash configure script has a `cache_file=/dev/null` so that could be set to a file containing those so no getenv needed

Reply:
This is good, very good. Now `hello` is down from 17 seconds to 6 seconds for whole build; `./configure` is faster by factor 10.

I didn't found way to fit it into `stdenv`, though. I think merging `autoconfCache` hook with `autoreconfHook` though. Thoughts?

Reply:
If I try to add `extraBuildInputs = [ prevStage.autoconfCache ]` into stdenv/linux, I get following error:

```
; nix-build -A stdenv
error: anonymous function at /home/kaction/devel/nix/nixpkgs/pkgs/build-support/fetchurl/boot.nix:5:1 called with unexpected argument 'postFetch'

at /home/kaction/devel/nix/nixpkgs/pkgs/build-support/fetchpatch/default.nix:31:1:

30| in
31| lib.throwIfNot (excludes == [] || includes == [])
| ^
32|   "fetchpatch: cannot use excludes and includes simultaneously"
(use '--show-trace' to show detailed location information)
```

It think it about pulling python (dependency of gnulib-tool) into bootstrap process.